spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -187,7 +241,32 @@ public FirestoreTemplate(FirestoreStub firestore, String parent) {
 		FirestorePersistentProperty idProperty = persistentEntity.getIdPropertyOrFail();
 		Object idVal = persistentEntity.getPropertyAccessor(entity).getProperty(idProperty);
 
-		return this.parent + "/" + persistentEntity.collectionName() + "/" + idVal.toString();
+		return buildResourceName(persistentEntity, idVal.toString());
+	}
+
+	private String buildResourceName(FirestorePersistentEntity<?> persistentEntity, String s) {
+		return this.parent + "/" + persistentEntity.collectionName() + "/" + s;
+	}
+
+	public String getIdValue(Object entity, FirestorePersistentEntity persistentEntity) {
```

Comment:
Can this also be private?

Otherwise this should be moved with other public methods and add javadoc.